ocd-2818.sql

```diff
@@ -0,0 +1,251 @@
+--
+-- OCD-2818 Change many-to-many mapping of users to roles to be one-to-one and migrate existing data
+--
+-- drop column
+ALTER TABLE openchpl.user DROP COLUMN IF EXISTS user_permission_id;
+-- add column, allow to be null for now until we have filled it in
+ALTER TABLE openchpl.user ADD COLUMN user_permission_id bigint;
+
+-- set permission for admin users
+UPDATE openchpl.user
+SET user_permission_id = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ADMIN')
+WHERE user_id IN 
+	(SELECT user_id 
+	 FROM openchpl.global_user_permission_map 
+	 WHERE user_permission_id_user_permission = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ADMIN'))
+AND user_permission_id IS NULL;
+
+-- set permissions for onc users
+UPDATE openchpl.user
+SET user_permission_id = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ONC')
+WHERE user_id IN 
+	(SELECT user_id 
+	 FROM openchpl.global_user_permission_map 
+	 WHERE user_permission_id_user_permission = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ONC'))
+AND user_permission_id IS NULL;
+
+-- set permissions for acb users
+UPDATE openchpl.user
+SET user_permission_id = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ACB')
+WHERE user_id IN 
+	(SELECT user_id 
+	 FROM openchpl.global_user_permission_map 
+	 WHERE user_permission_id_user_permission = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ACB'))
+AND user_permission_id IS NULL;
+
+-- set permissions for atl users
+UPDATE openchpl.user
+SET user_permission_id = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ATL')
+WHERE user_id IN 
+	(SELECT user_id 
+	 FROM openchpl.global_user_permission_map 
+	 WHERE user_permission_id_user_permission = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'ATL'))
+AND user_permission_id IS NULL;
+
+-- set permissions for cms_staff users
+UPDATE openchpl.user
+SET user_permission_id = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'CMS_STAFF')
+WHERE user_id IN 
+	(SELECT user_id 
+	 FROM openchpl.global_user_permission_map 
+	 WHERE user_permission_id_user_permission = (SELECT user_permission_id FROM openchpl.user_permission WHERE name = 'CMS_STAFF'))
+AND user_permission_id IS NULL;
+
+-- set permissions for onc_staff users (probably aren't any but just to make sure we have no null values in the table)
```

Comment:
We did get rid of them but since we soft delete it's technically still in the permissions table... I just thought best to add a line for them as well to be certain the user_permission_id wouldn't have any nulls. There aren't any ONC_STAFF users though so I could either remove this if you want or leave it. I don't think it's hurting anything but also probably not doing anything either.